pkg/websocket/websocket.go

```diff
@@ -157,46 +157,55 @@ func (ws *websocketClient) WriteJSON(v interface{}) error {
 }
 
 func (ws *websocketClient) readPump() {
-	defer ws.wg.Done()
-	ws.conn.SetReadDeadline(time.Now().Add(pongWait))
-	ws.conn.SetPongHandler(func(string) error { ws.conn.SetReadDeadline(time.Now().Add(pongWait)); return nil })
-	for {
-		_, message, err := ws.conn.ReadMessage()
-		if err != nil {
-			logrus.Error("websocket: readPump error:", err)
-			ws.wasDisconnected(err)
-			return
-		}
-		logrus.Debugf("websocket: readPump got msg: %s", message)
-		select {
-		case ws.read <- message:
-		default:
+	go func() {
+		ws.wg.Add(1)
```

Comment:
`ws.wg.Add(1)`should be done before running in gorouting (like i coded it before) to eliminate data race posibility.

Reply:
oops! I actually knew that